### PR TITLE
chore: Small cleanup to internal core utils

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR introduces some minor cleanup to improve consistency and structural semantics in `core`:
1. Export `WordArray` as module member so it can be re-exported directly by `libraryUtils.ts`
2. Extract `StrictUnion` to core root level types since it is not actually coupled to Auth (and will be used by Storage in future PRs)

#### Description of how you validated changes
```
$ yarn clean && yarn && yarn build && yarn test
```

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by david-mcafee — APPROVED** [review]

File: packages/core/src/types/utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+type UnionKeys<T> = T extends T ? keyof T : never;
+
+type StrictUnionHelper<T, TAll> = T extends any
+	? T & Partial<Record<Exclude<UnionKeys<TAll>, keyof T>, never>>
+	: never;
+
```
Comment (ashika112) [inline packages/core/src/types/utils.ts:9]: Nit: Should we add a comment to explain util function here?

**Review by israx — DISMISSED** [review]

File: packages/core/src/types/index.ts [hunk 2]
```diff
@@ -4,3 +4,4 @@
 export * from './core';
 export * from './errors';
 export * from './storage';
+export * from './utils';
```
Comment (ashwinkumar6) [inline packages/core/src/types/index.ts:7]: nit question,
Should these be updated to named exports later on to keep it uniform across all categories

**Review by ashwinkumar6 — DISMISSED** [review]
thank you!!

File: packages/core/src/types/index.ts (hunk 2 above)
Comment (cshfang) [reply packages/core/src/types/index.ts:7]: I'm not opposed to it but we do this in various other places as well - including other packages - so I'm not sure this is the right PR to do this in

**Review by israx — APPROVED** [review]

**Review by ashwinkumar6 — APPROVED** [review]

**Review by ashika112 — APPROVED** [review]

**Pull request merged**